lib/internal/validators.js

```diff
@@ -234,7 +230,10 @@ const validateAbortSignal = hideStackFrames((signal, name) => {
 });
 
 const validateFunction = hideStackFrames((value, name) => {
-  if (typeof value !== 'function')
+  if (inspect === undefined)
+    ({ inspect } = require('util'));
+
+  if (typeof value !== 'function' || inspect(value).startsWith('[class'))
     throw new ERR_INVALID_ARG_TYPE(name, 'Function', value);
 });
```

Comment:
This would make it possible to optionally allow classes. We can make use of this by calling `validateFunction(fn, 'fn', true)` inside timerify() instead of 
https://github.com/nodejs/node/blob/ccb8aae3932c13f33622203b2ffc5a33120e9d40/lib/internal/perf/timerify.js#L59
```js
const validateFunction = hideStackFrames((value, name, allowClass = false) => {
  if (typeof value === 'function') {
    if (allowClass === true)
      return;

    if (inspect === undefined)
      ({ inspect } = require('util'));

    if (!inspect(value).startsWith('[class'))
      return;
  }

  throw new ERR_INVALID_ARG_TYPE(name, 'Function', value);
});
```